packages/go/index.ts

```diff
@@ -94,332 +99,458 @@ export async function build({
   workPath,
   meta = {},
 }: BuildOptions) {
-  if (process.env.GIT_CREDENTIALS) {
-    debug('Initialize Git credentials...');
-    await initPrivateGit(process.env.GIT_CREDENTIALS);
-  }
+  const goPath = await getWriteableDirectory();
+  const srcPath = join(goPath, 'src', 'lambda');
+  const downloadPath = meta.skipDownload ? workPath : srcPath;
+  await download(files, downloadPath, meta);
 
-  if (process.env.GO111MODULE) {
-    console.log(`\nManually assigning 'GO111MODULE' is not recommended.
+  // keep track of file system actions we need to undo
+  // the keys "from" and "to" refer to what needs to be done
+  // in order to undo the action, not what the original action was
+  const undoFileActions: UndoFileAction[] = [];
+  const undoDirectoryCreation: string[] = [];
```

Comment:
This keeps a mostly clean user project space. It's not a complete cleanup where `go.mod` file changes are necessary, for example, but it's a lot better than it was.

Reply:
Silly question, but would it be better to just scan and copy the files into a tmp dir so that we don't need to touch the user's project files?

Reply:
There are pros and cons, but ultimately, I think we'd still want to do something like this. Maybe it's a virtualized file system, a full project copy, or a bunch of symlinks. I touched on this in the "Operating In-place" section of the description. I'm very open to alternative approaches to these shenanigans.

Reply:
I love the idea of a virtual FS! Symlinks work pretty well on posix platforms. Copying an entire project is ok if you're already copying and modifying many files anyways, otherwise maybe there's a JS implementation of rsync. In any case, my money is on the virtual filesystem.